confgenerator/logging_receivers.go

```diff
@@ -739,3 +740,47 @@ func (r LoggingReceiverSystemd) Pipelines(ctx context.Context) ([]otel.ReceiverP
 func init() {
 	LoggingReceiverTypes.RegisterType(func() LoggingReceiver { return &LoggingReceiverSystemd{} }, platform.Linux)
 }
+
+// LoggingCompositeReceiver represents a pipeline that consists of one log receiver & one or more log processors.
+type LoggingCompositeReceiver[R LoggingReceiverMixin, P LoggingMultiProcessorMixin] struct {
+	ConfigComponent     `yaml:",inline"`
+	MultiProcessorMixin P `yaml:",inline"`
```

Comment:
Why does this take a `MultiProcessorMixin` instead of just taking an `InternalLoggingProcessor` and deferring to that processor to construct the components?

Reply:
I created a `LoggingComponentMacro` struct to build a `LoggingCompositeReceiver` with the following two structs : 

https://github.com/GoogleCloudPlatform/ops-agent/blob/441081c4928b1bc66fb8858d2fbde8e8ccb4a13e/confgenerator/logging_macros.go#L26-L33

https://github.com/GoogleCloudPlatform/ops-agent/blob/441081c4928b1bc66fb8858d2fbde8e8ccb4a13e/confgenerator/logging_macros.go#L89-L93